### ISSUE TITLE
Xcode build for iOS fails under Appium Desktop.

### ISSUE THREAD

**Issue by tonyxyz** [appium/appium-desktop#291]
This is an issue with appium desktop. The problem does not occur running appium server from the command line. In both cases the appium version is reported as v1.6.5.

- [tick ] I have verified that my issue does not occur with Appium and should be investigated as an Appium Desktop issue

#### The problem

We run a cucumber test using the ruby cucumber gem and *on some developer machines* we see an error from Xcode as follow:

```
[XCUITest] Waiting up to 60000ms for WebDriverAgent to start
[JSONWP Proxy] Proxying [GET /status] to [GET http://localhost:8100/status] with no body
[Xcode] 2017-09-07 10:01:57.440 xcodebuild[1984:30002] [MT] DVTAssertions: ASSERTION FAILURE in /Library/Caches/com.apple.xbs/Sources/DVTFrameworks/DVTFrameworks-12154/DVTFoundation/MacroExpansion/DVTMacroDefinitionConditionSet.mm:154
[Xcode] Details:  stringRep should be a non-empty string, but it's an empty string
[Xcode] Object:   
[Xcode] Method:   +conditionSetFromStringRepresentation:getBaseMacroName:error:
[Xcode] Thread:   {number = 1, name = main}
[Xcode] Hints: 
[Xcode] 
[Xcode] Backtrace:
[Xcode]   0   -[DVTAssertionHandler handleFailureInMethod:object:fileName:lineNumber:assertionSignature:messageFormat:arguments:] (in DVTFoundation)
[Xcode]   1   _DVTAssertionHandler (in DVTFoundation)
[Xcode]   2   _DVTAssertionFailureHandler (in DVTFoundation)
[Xcode]   3   +[DVTMacroDefinitionConditionSet conditionSetFromStringRepresentation:getBaseMacroName:error:] (in DVTFoundation)
[Xcode]   4   -[DVTMacroDefinitionTable(XCMacroExpansionExtensions) _xc_setValue:forMacroName:errorHandler:] (in DevToolsCore)
[Xcode]   5   __108-[DVTMacroDefinitionTable(XCMacroExpansionExtensions) xc_setMacroNamesAndValuesFromDictionary:errorHandler:]_block_invoke (in DevToolsCore)
[Xcode]   6   __65-[__NSDictionaryI enumerateKeysAndObjectsWithOptions:usingBlock:]_block_invoke (in CoreFoundation)
[Xcode]   7   -[__NSDictionaryI enumerateKeysAndObjectsWithOptions:usingBlock:] (in CoreFoundation)
[Xcode]   8   -[DVTMacroDefinitionTable(XCMacroExpansionExtensions) xc_setMacroNamesAndValuesFromDictionary:errorHandler:] (in DevToolsCore)
[Xcode]   9   -[NSProcessInfo(XCMacroExpansionExtensions) xc_cachedEnvironmentAsMacroDefinitionTable] (in DevToolsCore)
[Xcode]  10   -[PBXTarget createMacroExpansionScopeWithBuildParameters:] (in DevToolsCore)
[Xcode]  11   -[PBXTarget cachedMacroExpansionScopeForBuildParameters:] (in DevToolsCore)
[Xcode]  12   -[PBXTarget expandedValueForString:forBuildParameters:] (in DevToolsCore)
[Xcode]  13   -[PBXNativeTarget fullProductNameForBuildParameters:] (in DevToolsCore)
[Xcode]  14   -[PBXNativeTarget fullProductNameForConfigurationNamed:] (in DevToolsCore)
[Xcode]  15   -[PBXTarget fullProductName] (in DevToolsCore)
[Xcode]  16   -[PBXReference setProducingTarget:] (in DevToolsCore)
[Xcode]  17   -[PBXTarget awakeFromPListUnarchiver:] (in DevToolsCore)
[Xcode]  18   -[PBXNativeTarget awakeFromPListUnarchiver:] (in DevToolsCore)
[Xcode]  19   -[NSArray makeObjectsPerformSelector:withObject:] (in CoreFoundation)
[Xcode]  20   -[PBXPListUnarchiver decodeRootObject] (in DevToolsCore)
[Xcode]  21   +[PBXProject projectWithFile:errorHandler:readOnly:] (in DevToolsCore)
[Xcode]  22   -[Xcode3Project initWithFilePath:extension:workspace:options:error:] (in DevToolsCore)
[Xcode]  23   __82+[IDEContainer _retainedContainerAtFilePath:fileDataType:workspace:options:error:]_block_invoke_2 (in IDEFoundation)
[Xcode]  24   _dispatch_client_callout (in libdispatch.dylib)
[Xcode]  25   _dispatch_barrier_sync_f_invoke (in libdispatch.dylib)
[Xcode]  26   DVTDispatchBarrierSync (in DVTFoundation)
[Xcode]  27   -[DVTDispatchLock performLockedBlock:] (in DVTFoundation)
[Xcode]  28   __82+[IDEContainer _retainedContainerAtFilePath:fileDataType:workspace:options:error:]_block_invoke (in IDEFoundation)
[Xcode]  29   __58-[DVTModelObjectGraph performBlockCoalescingModelChanges:]_block_invoke (in DVTFoundation)
[Xcode]  30   -[DVTModelGraphTransactionScope performTransaction:] (in DVTFoundation)
[Xcode]  31   -[DVTModelObjectGraph performBlockCoalescingModelChanges:] (in DVTFoundation)
[Xcode]  32   +[IDEContainer _retainedContainerAtFilePath:fileDataType:workspace:options:error:] (in IDEFoundation)
[Xcode]  33   +[IDEContainer retainedContainerAtFilePath:fileDataType:workspace:error:] (in IDEFoundation)
[Xcode]  34   -[IDEWorkspace _configureWrappedWorkspaceWithError:] (in IDEFoundation)
[Xcode]  35   __82+[IDEContainer _retainedContainerAtFilePath:fileDataType:workspace:options:error:]_block_invoke (in IDEFoundation)
[Xcode]  36   +[IDEContainer _retainedContainerAtFilePath:fileDataType:workspace:options:error:] (in IDEFoundation)
[Xcode]  37   +[IDEContainer retainedWrappedWorkspaceForContainerAtFilePath:fileDataType:error:] (in IDEFoundation)
[Xcode]  38   -[Xcode3CommandLineBuildTool _resolveInputOptionsWithTimingSection:] (in Xcode3Core)
[Xcode]  39   -[Xcode3CommandLineBuildTool run] (in Xcode3Core)
[Xcode]  40  0x000000010abd22af (in xcodebuild)
[Xcode]  41   start (in libdyld.dylib)
[Xcode] 
[XCUITest] xcodebuild exited with code 'null' and signal 'SIGABRT'
```

We start the test with these caps:

```appium_config = {
    caps: {
      platformName: :ios,
      deviceName: 'iPad Air',
		  platformVersion: '10.3',
		  app: app_path,
		  automationName: 'XCUITest',
			showXcodeLog: true
    },
   appium_lib: {
      wait_timeout: 30
    }
  }
```

## Environment

- I am running Appium Desktop version 1.2.0-beta.3 (also occurs in 1.1.1)
- macOS 10.12.6
- xcode-select version 2347
- xcrun version 32
- Xcode 8.3.3 (8E3004b)

**Comment by jlipps** [comment]
@tonyxyz since you say this is not totally reproducible, how sure are you that it only happens in appium desktop? have you run regular appium sessions enough times to be reasonably certain?

i'm asking because i can't think of anything that would cause this error from appium desktop itself. appium desktop is simply running appium under the hood, after all.

@mykola-mokhnach have you seen this kind of error when building WDA before?

**Comment by mykola-mokhnach** [comment]
I bet this is some internal Xcode bug. Update/reinstall Xcode might help

**Comment by tonyxyz** [comment]
Yes it definitely does not occur with Appium running in a tty. One observation is a colleague had different problems with the Appium Desktop WDA compilation - this turned out to be caused by running the program from an unexpected location, ~/Downloads, the result of this was that the program had unexpected permission attributes - which resulted in 'read only file system error' being thrown from within the build. This appeared to be caused by ACLs rather than POSIX level file permission stuff. We fixed it by making sure he was running Appium Desktop from Applications. This was different to the error posted here but shows the kind of unexpected subtle problems that can happen.

We already tried a reinstall of Xcode but will try a really thorough clean and reinstall next week.
